istioctl/cmd/workload.go

```diff
@@ -378,9 +381,12 @@ func createSidecarEnv(internalIP string, externalIP, dir string) error {
 	return os.WriteFile(filepath.Join(dir, "sidecar.env"), []byte(mapToString(sidecarEnv)), filePerms)
 }
 
-func generateSidecarEnvAsMap(internalIP string, externalIP string) map[string]string {
+func generateSidecarEnvAsMap(internalIP string, externalIP string, rev string) map[string]string {
 	sidecarEnv := make(map[string]string)
 
+	if addr := revisionedIstiodAddress(rev); addr != "" {
+		sidecarEnv["PILOT_ADDRESS"] = addr
```

Comment:
Why do we need this if we have the EXPLICITLY_SET_PILOT_ADDRESS thing?

Reply:
We don't -- I didn't revert it correctly.